rubin_sim/scheduler/basis_functions/basis_functions.py

```diff
@@ -193,6 +194,29 @@ def check_feasibility(self, conditions):
         return result
 
 
+class FilterDistBasisFunction(BaseBasisFunction):
+    """Track filter distribution, increase reward as fraction of observations in
+    specified filter drops.
+    """
+
+    def __init__(self, filtername="r"):
+        super(FilterDistBasisFunction, self).__init__(filtername=filtername)
+
+        self.survey_features = {}
+        # Count of all the observations
+        self.survey_features["n_obs_count_all"] = features.NObsCount(filtername=None)
+        # Count in filter
+        self.survey_features["n_obs_count_in_filt"] = features.NObsCount(
+            filtername=filtername
+        )
+
+    def _calc_value(self, conditions, indx=None):
+        result = self.survey_features["n_obs_count_all"].feature / (
+            self.survey_features["n_obs_count_in_filt"].feature + 1
+        )
+        return result
```

Comment:
For the twilight inner solar system survey we can't use the regular footprint basis function because we don't want to enforce spatial uniformity of the observations. We just want them on the ecliptic, we don't care if parts of the ecliptic don't happen to get observed. So the footprint basis function ends up with zero weight, it's just there as an effective mask, so it can't regulate filter distribution.

Reply:
Ah, thanks! That's a good reason.